Lib/email/message.py

```diff
@@ -286,8 +287,11 @@ def get_payload(self, i=None, decode=False):
         if i is not None and not isinstance(self._payload, list):
             raise TypeError('Expected list, got %s' % type(self._payload))
         payload = self._payload
+        cte = self.get('content-transfer-encoding', '')
+        if isinstance(cte, headerregistry.ContentTransferEncodingHeader):
+            cte = cte.cte
         # cte might be a Header, so for now stringify it.
-        cte = str(self.get('content-transfer-encoding', '')).lower()
+        cte = str(cte).strip().lower()
```

Comment:
This line only really applies if the header *wasn't* pre-parsed by the HeaderRegistry (ie: it is old-api backward compatibility code at this point in time) so I'd put it in an 'else' clause of the above 'if'.

Which brings up the point that we need an additional test for the compat32 policy that blanks get stripped, since in the existing code they do not.